pootle/apps/virtualfolder/management/commands/add_vfolders.py

```diff
@@ -0,0 +1,128 @@
+
+        logging.info("\nSuccesfully parsed '%s'\n", vfolders_filename)
+
+        added_count = 0
+        updated_count = 0
+
+        for vfolder_item in vfolders:
+            vfolder_item['name'] = vfolder_item['name'].lower()
+
+            # Put all the files for each virtual folder as a list and save it
+            # as its filter rules.
+            if 'files' in vfolder_item:
+                vfolder_item['filter_rules'] = ','.join(vfolder_item['files'])
+                del vfolder_item['files']
+            else:
+                vfolder_item['filter_rules'] = ''
+
+            # Now create or update the virtual folder.
+            try:
+                # Retrieve the virtual folder if it exists.
+                vfolder = VirtualFolder.objects.get(
+                    name=vfolder_item['name'],
+                    location=vfolder_item['location'],
+                )
+            except VirtualFolder.DoesNotExist:
+                # If the virtual folder doesn't exist yet then create it.
+                vfolder = VirtualFolder(**vfolder_item)
+                vfolder.save()
+                added_count += 1
+            else:
+                # Update the already existing virtual folder.
+                changed = False
+
+                if vfolder.filter_rules != vfolder_item['filter_rules']:
+                    vfolder.filter_rules = vfolder_item['filter_rules']
+                    changed = True
+                    logging.info("Filter rules for virtual folder '%s' will "
+                                 "be changed.", vfolder.name)
+
+                if ('priority' in vfolder_item and
+                    vfolder.priority != vfolder_item['priority']):
+
+                    vfolder.priority = vfolder_item['priority']
+                    changed = True
+                    logging.info("Priority for virtual folder '%s' will be "
+                                 "changed to %f.", vfolder.name,
+                                 vfolder.priority)
+
+                if ('is_browsable' in vfolder_item and
+                    vfolder.is_browsable != vfolder_item['is_browsable']):
+
+                    vfolder.is_browsable = vfolder_item['is_browsable']
+                    changed = True
+                    logging.info("is_browsable status for virtual folder '%s' "
+                                 "will be changed.", vfolder.name)
+
+                if ('description' in vfolder_item and
+                    vfolder.description.raw != vfolder_item['description']):
+
+                    vfolder.description = vfolder_item['description']
+                    changed = True
+                    logging.info("Description for virtual folder '%s' will be "
+                                 "changed.", vfolder.name)
+
+                if changed:
+                    vfolder.save()
+                    updated_count += 1
+
+        logging.info("\nSucessfully added %d virtual folders.\nSuccessfully "
```

Comment:
There's a typo (*sucessfully*), but besides, it looks easier to parse an output such as:
```
Added: M
Updated: N
Unchanged: P
```